app/controllers/DigitalSubscription.scala

```diff
@@ -105,20 +105,36 @@ class DigitalSubscription(
     )
   }
 
+  sealed trait CreateDigitalSubscriptionError {
+    def asString: String
+  }
+  case class ServerError(message: String) extends CreateDigitalSubscriptionError {
+    override def asString: String = message
+  }
+  case class RequestValidationError(message: String) extends CreateDigitalSubscriptionError {
+    override def asString: String = message
```

Comment:
do you need the asString as well as message or can you just call it message (and remove asString from the subclasses)?
or vice versa